src/sync.rs

```diff
@@ -677,6 +693,40 @@ impl<T: Copy> LockFreeFrozenVec<T> {
         let local_index = index - prior_total_buffer_size(buffer_idx);
         unsafe { *buffer_ptr.add(local_index) }
     }
+
+    /// Run a function on each buffer in the vector.
+    ///
+    /// ## Arguments
+    /// - `func`: a function that takes a pointer to the buffer, the buffer size, and the buffer index
+    ///
+    /// ## Safety
+    /// `func` is provided with the raw constant pointer to the buffer,
+    /// however, the pointer is expected to be valid as the null check is done before calling `func`.
+    /// To access the data in the buffer, make sure the buffer size (the second argument) is respected.
+    ///
+    pub unsafe fn for_each_buffer(&self, mut func: impl FnMut(*const T, usize, usize)) {
```

Comment:
I made this function `pub`, but if you think the underlying implementation might change, I can make it private.

Reply:
Having it pub seems fine to me.

You can avoid the `Fn`/`FnMut` dance by changing the function to return an `impl Iterator<(*const T, usize, usize)>`.